test/studies/diffusion/heat_2d_dist.chpl

```diff
@@ -0,0 +1,51 @@
+use StencilDist;
+
+// setup simulation parameters
+config const xLen : real = 2,
+             yLen : real = 2,
+             nx = 31,
+             ny = 31,
+             nt = 50,
+             sigma = 0.25,
+             nu = 0.05;
+
+const dx : real = xLen / (nx - 1),
+      dy : real = yLen / (ny - 1),
+      dt : real = sigma * dx * dy / nu;
+
+// define a 2D domain and subdomain
+const dom = {0..<nx, 0..<ny},
+      domInner = dom.expand(-1),
+      DOM = dom dmapped Stencil(domInner, fluff=(1, 1)),
+      DOMINNER : subdomain(DOM) = DOM[domInner];
```

Comment:
I'd definitely drop the `: subdomain(DOM)` from here, as `DOM[someDomain]` is a subdomain of DOM by definition (and you haven't used type specifications for any of the other domain declarations).  Also, as mentioned above, `subdomain()` is not an awesome feature at present.